server/src/main/java/org/elasticsearch/cluster/coordination/PublicationTransportHandler.java

```diff
@@ -225,7 +220,7 @@ public PublicationContext newPublicationContext(ClusterStatePublicationEvent clu
 
     private ReleasableBytesReference serializeFullClusterState(ClusterState clusterState, DiscoveryNode node) {
         final Version nodeVersion = node.getVersion();
-        final BytesStreamOutput bytesStream = new ReleasableBytesStreamOutput(bigArrays);
+        final BytesStream bytesStream = transportService.newNetworkBytesStream();
```

Comment:
I guess my main question here is there are reason we don't just specifically classify that this is a `RecyclerBytesStreamOutput`? I like to be explicit that this HAS to be closed, as there are plenty of places we do not close `BytesStream` instances. And it avoids the need to change the signature of `BytesStream`. You can still make the `default` `TransportService` method return a non recycling instance.  Also, presumably `RecyclerBytesStreamOutput` is just going to be come the default `BytesStreamOutput` [soon](https://github.com/elastic/elasticsearch/pull/80388). Not that impacts any of your work here.

Reply:
`new RecyclerBytesStreamOutput(new BytesRefRecycler(PageCacheRecycler.NON_RECYCLING_INSTANCE))` is a non-recycling instance. Can even add a static constant to `BytesRefRecycler.NON_RECYCLING_INSTANCE`.

Reply:
Primarily because I didn't see that I could use `BytesRefRecycler` to bridge the gap at that level instead. (Also to avoid having to replace the `DelegatingBigArrays` thing in `AbstractCoordinatorTestCase`, although that wasn't so bad after all)